lib/models/infrastructure.rb

```diff
@@ -146,9 +145,20 @@ def submit_create
     end
     self
   end
-
+  
   def submit_update
-    response = hyper_client.put(infrastructure_url(infrastructure_id: remote_id), api_format.merge(status: 'Active'))
+    logger.info "Updating infrastructure #{name} in UC6 API"
+    begin
+      response = hyper_client.put(infrastructure_url(infrastructure_id: remote_id), api_format.merge(status: 'Active'))
+      response_json = response.json
+      if (response.present? && response.code == 200 && response_json['id'].present?)
+        self.record_status = 'verified_update'
+      end
+    rescue
+      logger.error "Error updating infrastructure '#{name} in UC6"
+      raise
```

Comment:
Maybe we may need to raise the specific exception here, so we could rescue with RuntimeError => e and then raise "e"